tests/test_discount.py

```diff
@@ -0,0 +1,31 @@
+from pywr.parameters import Parameter, load_parameter
+import datetime
+
+class DiscountFactorParameter(Parameter):
+    """Parameter that returns the current discount factor based on discount rate and a base year.
+
+    Parameters
+    ----------
+    discount_rate : discount rate fixed across planning period
+    base_year : base year in 'yyyy' format
+
+    """
+
+    def __init__(self, model, discount_rate, base_year):
+        super(DiscountFactorParameter, self).__init__(model)
+        self.discount_rate = discount_rate
+        self.base_year = base_year
+
+    def value(self, discount_rate, base_year):
+        ct = self.model.timestepper.current #[0:4]
+        cy = ct.year
+        cyi = cy - self.base_year
+        return 1/pow((1+self.discount_rate), cyi) # return discount factor of current year
+
+    @classmethod
+    def load(cls, model, data):
```

Comment:
This could be simplified:

```python
    @classmethod
    def load(cls, model, data):
        return cls(model, **data)
```

The current code doesn't support  these attributes being parameters. The above will mean you can provide an non-parameter constant values only. If you want to support discount rate being a parameter then it needs a bit more treatment in `__init__` and `value`. I would say start with this, and add support for this later.

Reply:
Thanks James, I am not sure the changes I made allows the discount rate being a parameter. I added `**kwargs` to  `__init__` and `value` and `**data` to the `return` function. 
Did you suggest that I removed `discount_rate` and  `base_year` from the `return` function and just use `**data`? If so, I'm not sure how to do this but I could have a go.

Reply:
Sorry I missed replying to this. The main question is whether you want to support the discount rate varying as a `Parameter`. The problem is the current implementation in `value` is assuming a fixed discount rate and it also does not deal with parameter dependency properly in the `__init__` method (see: https://pywr.github.io/pywr-docs/master/extending_pywr/extending_pywr_parameters.html#dependency-on-other-parameters)

By using 

```json
@classmethod
    def load(cls, model, data):
        return cls(model, **data)
```

It will pass the values in the JSON straight to `__init__` with no conversion. This is the correct way to do it because your code is treating `self.discount_rate` as a `float` and not a `Parameter` instance.

Obviously this means implementation is limited to a fixed discount rate at the moment, but let's get that right first.